Decide/src/main/java/DD2480/decide/CMVHandler.java

```diff
@@ -83,14 +81,34 @@ private boolean licTwo() {
 	}
 
 	private boolean licThree() {
-		// TODO Auto-generated method stub
-		return false;
+        for (int index = 0; index < dataPoints.length - 2; index++) {
+            if (GeometryHelper.area(dataPoints[index], dataPoints[index + 1], dataPoints[index + 2]) > parameters.area1)
+                return true;
+        }
+        return false;
 	}
 
 	private boolean licFour() {
-		// TODO Auto-generated method stub
-		return false;
-	}
+        for (int index = 0; index < (dataPoints.length - parameters.qPts + 1); index++) {
+            boolean[] quads = {false, false, false, false};
+
+            for (int i = index; i < parameters.qPts; i++) {
```

Comment:
It seems to me that you only check only on set (the first) of Q_PTS. But if Numpoints is bigger than Q_PTS then there are more set of points that has to be checked.